mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -413,11 +415,40 @@ fn update_failfast_cb(
 ) -> i32 {
     let old_value = channel.fail_fast;
 
+    // Increment fail-fast counter and close I/O handles for target device.
     if ctx.increment {
+        let target_device = ctx.failed_device.as_ref().expect(
+            "Device name must be provided when incrementing fail-fast counter",
+        );
+
         channel.fail_fast = channel
             .fail_fast
             .checked_add(1)
             .expect("Fail-fast counter overflow");
+
+        // Close I/O handle to cancel all active I/O for target device.
+        [&mut channel.readers, &mut channel.writers]
+            .iter_mut()
+            .for_each(|c| {
+                c.iter_mut().for_each(|h| {
+                    if h.get_device().device_name().eq(target_device) {
+                        match h.close() {
+                            Err(e) => {
+                                error!(
+                                    "{} failed to close device handle handle: {:?}",
+                                    target_device, e
+                                );
+                            },
+                            _ => {
+                                info!(
```

Comment:
`nvmx::handle::close()` already logs an `info` message so suggest removing this one.